Sources/Workspace/Destination.swift

```diff
@@ -111,35 +111,41 @@ public struct Destination: Encodable, Equatable {
             }
             sdkPath = AbsolutePath(sdkPathStr)
         }
+#else
+        sdkPath = nil
+#endif
 
         // Compute common arguments for clang and swift.
         var extraCCFlags: [String] = []
         var extraSwiftCFlags: [String] = []
+#if os(macOS)
         if let sdkPaths = Destination.sdkPlatformFrameworkPaths(environment: environment) {
             extraCCFlags += ["-F", sdkPaths.fwk.pathString]
             extraSwiftCFlags += ["-F", sdkPaths.fwk.pathString]
             extraSwiftCFlags += ["-I", sdkPaths.lib.pathString]
             extraSwiftCFlags += ["-L", sdkPaths.lib.pathString]
         }
+#endif
+
+#if !os(Windows)
+        extraCCFlags += ["-fPIC"]
+#endif
+
+        var extraCPPFlags: [String] = []
+#if os(macOS)
+        extraCPPFlags += ["-lc++"]
+#else
+        extraCPPFlags += ["-lstdc++"]
```

Comment:
This is also incorrect for Windows: windows generally uses msvcrprt, but you need to know what configuration you are building in as the debug and release mode builds are ABI incompatible.  You should either link `msvcprt` or `msvcprtd` assuming dynamic linking.  If you are statically linking, then the library names have alternate names: `libcpmt` and `libcpmtd`.  Really the ideal way to handle this is to use the language driver to drive the link process for C++ code which should work everywhere with the proper default behaviour (though some flags should get plumbed through for `-stdlib=libc++` on Linux/Android, `-stdlib=libc++` on Windows and `/MT`, `/MTd`, `/MD`, `/MDd` on Windows).

Reply:
@compnerd thanks for the heads up!
The CPP flags logic wasn't changed in any meaningful way though, I just extracted the `extraCPPFlags` array to have a single `Destination` initializer. I didn't intend to modify the CPP flags as they are out of scope of this PR.

Reply:
Ah, okay, sounds like there is still a lot to be done here to make this correct.